### PR TITLE
chore(ui): migrate Box component to Typescript

### PR DESCRIPTION
**Pull request by guoda-puidokaite**
# Summary

- Migrated the `Box` component to `TypeScript`, enhancing code quality, maintainability, and scalability.
- Utilised static typing to minimize runtime errors and improve the development experience.
- Enhanced unit tests, covering more edge cases.
- Improved documentation with clearer descriptions of the component and its props.

# Changes Made

**TypeScript Migration**:

- Converted `Box` component files to `TypeScript`.
- Added comprehensive type annotations.

**Refactoring**:

- Made necessary adjustments to address TypeScript-detected issues.
- Further refactored for cleaner and more understandable code.

**Testing**:

- Added more Edge Cases:
  - Rendering Without Children: Ensured the Box component renders correctly without children.
  - Handling Additional Attributes: Ensured the Box component handles additional attributes without errors.
- Organised tests by functionality for better readability and easier maintenance.
- Verified component behaviour using `Storybook`.
- Updated tests to use `vitest`.

**Cleanup:**

- Removed deprecated code. It wasn't needed for any `.js` components.

**Documentation:**

- Updated `Storybook` documentation.

Screenshot: Deprecated `.js` code on the left, migrated `.ts` code on the right (**see notes**)

Screenshot 2024-10-17 at 15 22 41

**Release:**

- Generated a `changeset` file, mentioning the migration and bug fix.

# Related Issues

- https://github.com/cloudoperators/juno/issues/247

# Testing Instructions

1. Run `Storybook`
2. Run `Box` tests

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

**Comment by guoda-puidokaite** [comment]
Closes https://github.com/cloudoperators/juno/issues/214

File: packages/ui-components/src/components/Box/Box.component.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const boxPadding = `
+  jn-py-1
+  jn-px-2
+`
+
+export interface BoxProps {
+  /**
+   * Child elements to be rendered inside the Box.
+   */
+  children?: React.ReactNode
```
Comment (guoda-puidokaite) [inline packages/ui-components/src/components/Box/Box.component.tsx:26]: This prop should probably be mandatory, as without content the Box component has no purpose?
There's no other props to pass content.

File: packages/ui-components/src/components/Box/Box.component.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  /**
+   * If true, padding is removed.
+   */
+  unpad?: boolean
```
Comment (guoda-puidokaite) [inline packages/ui-components/src/components/Box/Box.component.tsx:30]: FYI about [unpad prop naming](https://github.com/cloudoperators/juno/issues/214#issuecomment-2419539515)

**Review by barsukov — APPROVED** [review]
LGTM 🎉

**Review by TilmanHaupt — APPROVED** [review]

**Review by edda — APPROVED** [review]
